var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -76,12 +76,20 @@ class Esmf(MakefilePackage):
     )
     variant("debug", default=False, description="Build with debugging symbols and options enabled")
     variant("shared", default=True, description="Build shared library")
+
     # 'esmf_os', 'esmf_comm', 'esmf_pio' variants allow override values for their corresponding
     # build environment variables. Documentation, including valid values, can be found at
     # https://earthsystemmodeling.org/docs/release/latest/ESMF_usrdoc/node10.html#SECTION000105000000000000000
     variant("esmf_os", default="auto", description="Override for ESMF_OS variable")
     variant("esmf_comm", default="auto", description="Override for ESMF_COMM variable")
     variant("esmf_pio", default="auto", description="Override for ESMF_PIO variable")
+    # Set the 'snapshot' variant any time a beta snapshot is used in order to obtain
+    # correct module name behavior for MAPL.
+    variant(
+        "snapshot",
+        default="none",
+        description="Named variant for snapshots versions (e.g., 'b09')",
```

Comment:
I don't see this variant being passed to the build process.  Am I missing something?

Reply:
It's a workaround for the fact that in modules.yaml, spack doesn't distinguish between, say, "@1.2.3" and "@1.2.3b09". I discussed this at some length with Todd once upon a time and there wasn't really another good solution. So this is essentially a dummy variant that let's us distinguish ESMF's "b??" subversions when we customize the modulefile names of one of our dependencies (MAPL).

Reply:
On question to consider - and I do not know the answer - is how much longer will anything depend on a specific snapshot tag like v8.3.0b09? I think it was unfortunate that some models started depending on that specific snapshot tag, but hopefully that is a thing of the past now?

Reply:
If I was the one to make this decision, I would say that any submitted ufs-weather-model code must use an official release of ESMF. But as long as UFS WM developers want to test with a beta snapshot (and have the mapl dependency be built with it, too), and spack(-stack) is used to install that version, this code needs to stay.

Reply:
It's ok of course to leave this out of mainline spack and keep that "feature" in our fork, just one more difference to carry around.

Reply:
Thanks @climbfuji for clarifying. I have no issue with this "feature" moving into the mainline spack ESMF package. Generally I think the fewer differences between your fork and the mainline, the better.

Reply:
I don't want to hold up the PR on this but would like to understand the notion better: the module file should be able to access the version of dependencies (e.g. `8.3.0b09`). I can see how you might not want to alter all module file names just to account for this.

That being said I think this variant could be named more-clearly, like:

```
variant(ufs-weather-model)
```

And furthermore if it's a dummy variant which derives from the version, it might also be useful to encode a conflict

```
conflicts(+ufs-weather-model, when=)
```

admittedly this would be difficult to encode given that we don't support `not`.